src/RecursiveIterableAggregate.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+declare(strict_types=1);
+
+namespace loophp\iterators;
+
+use Closure;
+use Generator;
+use IteratorAggregate;
+
+/**
+ * @template TKey
+ * @template T
+ *
+ * @implements IteratorAggregate<TKey, T>
+ */
+class RecursiveIterableAggregate implements IteratorAggregate
+{
+    /**
+     * @param iterable<TKey, T> $iterable
+     * @param (Closure(T, TKey, iterable<TKey, T>): iterable<TKey, T>) $closure
+     */
+    public function __construct(private iterable $iterable, private Closure $closure) {}
+
+    /**
+     * @return Generator<TKey, T>
+     */
+    public function getIterator(): Generator
+    {
+        yield from $this->flatten($this->iterable);
+    }
+
+    /**
+     * @param iterable<TKey, T> $items
+     *
+     * @return Generator<TKey, T>
+     */
+    private function flatten(iterable $items): Generator
+    {
+        foreach ($items as $key => $value) {
+            yield $key => $value;
+
+            yield from $this->flatten(($this->closure)($value, $key, $this->iterable));
```

Comment:
I'm trying to avoid recursive things in PHP, it's a nightmare for performance.

One of my last commit in this project removed recursivity to have a huge gain, see https://github.com/loophp/iterators/commit/1119d7c069da02221cb9150819d30c44758ea4f9

Reply:
Good point. I've changed the implementation to use loop instead of recursive calls.